tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -0,0 +1,246 @@
+      raise ValueError('Cannot calculate cross-products in {} dimensions for '
+                       'samples of rank {}'.format(event_ndims, len(shape)))
+    if event_ndims > 13:
+      raise ValueError('`event_ndims` over 13 not supported')
+    self.shape = shape
+    self.event_ndims = event_ndims
+    self.dtype = dtype
+
+  def initialize(self):
+    """Initializes a `RunningCovarianceState` using previously defined metadata.
+
+    Returns:
+      state: `RunningCovarianceState` representing a stream of no inputs.
+    """
+    # we need a secondary `RunningCovarianceState` so that future calls to
+    # `update` are compatible with `tf.while_loop`. Namely, we need to
+    # somehow store the `event_ndims` and `dtype` without explicitly passing or
+    # returning them from a `tf.while_loop`
+    if self.event_ndims == 0:
+      extra_ndims_shape = ()
+    else:
+      extra_ndims_shape = self.shape[-self.event_ndims:]
+    init_state = RunningCovarianceState(
+        num_samples=tf.zeros((), dtype=self.dtype),
+        mean=tf.zeros(self.shape, dtype=self.dtype),
+        comoment=tf.zeros(self.shape + extra_ndims_shape, dtype=self.dtype),
+    )
+    return RunningCovarianceState(**init_state._asdict())
+
+  def update(self, state, new_sample, axis=None):
+    """Update the `RunningCovarianceState` with a new sample.
+
+    The update formula is from Philippe Pebay (2008). This implementation
+    supports both batched and chunked covariance computation. A "batch" is the
+    usual parallel computation, namely a batch of size N implies N independent
+    covariance computations, each stepping one sample (or chunk) at a time. A
+    "chunk" of size M implies incorporating M samples into a single covariance
+    computation at once, which is more efficient than one by one.
+
+    To further illustrate the difference between batching and chunking, consider
+    the following example:
+
+    ```python
+    import tensorflow as tf
+    import tensorflow_probability as tfp
+    # treat as 3 samples from each of 5 independent vector random variables of
+    # shape (2,)
+    sample = tf.ones((3, 5, 2))
+    running_cov = tfp.experimental.stats.RunningCovariance(
+        (5, 2), event_ndims=1)
+    state = running_cov.initialize()
+    state = running_cov.update(state, sample, axis=0)
+    final_cov = running_cov.finalize(state)
+    final_cov.shape # (5, 2, 2)
+    ```
+
+    Args:
+      state: `RunningCovarianceState` that represents the current state of
+        running statistics.
+      new_sample: Incoming sample with shape and dtype compatible with those
+        used to form the `RunningCovarianceState`.
+      axis: If chunking is desired, this is an integer that specifies the axis
+        with chunked samples. For individual samples, set this to `None`. By
+        default, samples are not chunked (`axis` is None).
+
+    Returns:
+      state: `RunningCovarianceState` with updated calculations.
+
+    References:
```

Comment:
The format here is wrong (see the STYLE_GUIDE.md), it should be:

```
#### References

[1]: Tony Finch. Incremental calculation of weighted mean and variance.
     _Technical Report_, 2009. Blah blah

Reply:
Not exactly sure which part is formatted wrong (could you point me towards it)? Is it the fact that leaving _Technical Report_ is sufficient?

Reply:
References should be a markdown heading, and it should use hanging alignment.

```
#### References

[1]: Philippe Pebay. Formulas for Robust, One-Pass Parallel Computation of
     Covariances and Arbitrary-Order Statistical Moments. _Technical Report
     SAND2008-6212_, 2008.
     https://prod-ng.sandia.gov/techlib-noauth/access-control.cgi/2008/086212.pdf
```

Reply:
Oops, I was looking at all the wrong places haha. Will fix